lago/cmd.py

```diff
@@ -899,7 +900,28 @@ def create_parser(cli_plugins, out_plugins):
     return parser
 
 
+def sigterm_handler(signum, frame):
+    """
+    Catch SIGTERM and call "sys.exit" which raises "SystemExit" exception.
+    This will trigger all the cleanup code defined in ContextManagers
+    and "finally" statements.
+
+    For more details about the arguments see "signal" documentation.
+
+    Args:
+        signum(int): The signal's number
+        frame(frame): The current stack frame, can be None
+    """
+
+    LOGGER.debug('SIGTERM was caught')
+    sys.exit(128 + signum)
+
+
 def main():
+
+    # Trigger cleanup on SIGTERM
+    signal(SIGTERM, sigterm_handler)
```

Comment:
I think you should catch SIGHUP as well just for cases where Lago is used on a remote host and connection closes.